### DISCUSSION TITLE
How I can copy folder across different vims?

### DISCUSSION THREAD

**Discussion by Demianeen** [nvim-neo-tree/neo-tree.nvim#1260] [Q&A] [answered]
I often need to copy files across multiple projects, so, multiple neovim instances. When I try to copy files in one neovim and paste them into another, that's not working. Is it possible?

**Comment by miversen33** [comment]
IIRC Neo-tree is not "instance aware", meaning it does not "talk" to other neo-tree instances. Which also means that something like this is not currently possible.

I could be incorrect and maybe there is some feature baked into Neo-tree that might enable this kind of behavior but if I am not wrong, this would require some work handling "paste" input.

I wouldn't say this is "impossible". It might be possible to accomplish something like this using a custom URI on copy (something like `neo-tree-file:///myfile.txt` instead of `file:///myfile.txt` or simply `myfile.txt`) and then interpreting the URI on paste event but even Neo-tree would need "outside world" help in configuring your system clipboard to shove what it yanks into the system clipboard.

Basically "No" unless you want to hack something together. There are events and hooks you can potentially use to implement this behavior yourself but (unless I am mistaken), it is highly unlikely this kind of functionality is already available with Neo-tree.

If you are interested in tackling this for yourself, take a peek at the "mappings" table you can provide during neo-tree configuration. 
`:h neo-tree-custom-mappings`

The maintainer just dropped a mapping snippet [here](https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/1257#discussioncomment-7811932) that does some (completely unrelated) complex logic using mappings, maybe between it and the [source](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/sources/filesystem/commands.lua) you can work your way to what you want :)

If you do decide to tackle this and have questions, feel free to `@` me with them :) I wouldn't say I am the most knowledgable about neo-tree but I have enough experience with the ole 'Hackity Schmakity' with neo-tree that I can likely help and I am mildly intrigued with this request. Not enough to do it but enough to provide guidance if its requested

**Comment by cseickel** [comment, marked as answer]
There is actually an issue for handling copy paste between two windows within the same neovim instance that will probably never happen: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/168

Copy across different neovim instances would be another thing altogether. If they are at least on the same computer, I can see implementing the clipboard as a file somewhere in the user's `~/.local/` folder which wouldn't be that difficult. I wouldn't do it officially though because I wouldn't want to support the feature.

**Reply by Demianeen** [reply]
@cseickel hi!
I think about how I started to do this:
1. I just copy path in one nvim
2. use cp command in other terminal with . as path. 
Do you think it is possible to do a command that will run cp from that folder based on the copied path from the clipboard to the focused dir? Or do you think it is just easier without it?
